src/agent/src/device/mod.rs

```diff
@@ -1192,9 +1192,9 @@ mod tests {
         );
         spec.set_annotations(Some(annotations));
 
-        // create a file in /tmp/cdi with nvidia.json content
-        let cdi_dir = PathBuf::from("/tmp/cdi");
-        let cdi_file = cdi_dir.join("kata.json");
+        // Create a temporary directory for CDI￼
+        let temp_dir = tempdir().expect("Failed to create temporary directory");
+        let cdi_file = temp_dir.path().join("kata.json"); // create a file in /tmp/cdi with nvidia.json content
```

Comment:
The code is trivial enough that it doesn't need an extra comment IMHO. Also the comment is slightly wrong as it is still mentioning `/tmp/cdi` :wink:

Reply:
Yeah copy and waste